connectors/src/resources/slack_configuration_resource.ts

```diff
@@ -0,0 +1,216 @@
+import type { ModelId, Result } from "@dust-tt/types";
+import { Err, Ok } from "@dust-tt/types";
+import type { Attributes, ModelStatic, Transaction } from "sequelize";
+
+import {
+  SlackBotWhitelistModel,
+  SlackChannel,
+  SlackConfigurationModel,
+  SlackMessages,
+} from "@connectors/lib/models/slack";
+import logger from "@connectors/logger/logger";
+import { BaseResource } from "@connectors/resources/base_resource";
+import type { ReadonlyAttributesType } from "@connectors/resources/storage/types";
+
+// Attributes are marked as read-only to reflect the stateless nature of our Resource.
+// This design will be moved up to BaseResource once we transition away from Sequelize.
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface SlackConfigurationResource
+  extends ReadonlyAttributesType<SlackConfigurationModel> {}
+export class SlackConfigurationResource extends BaseResource<SlackConfigurationModel> {
+  static model: ModelStatic<SlackConfigurationModel> = SlackConfigurationModel;
+
+  constructor(
+    model: ModelStatic<SlackConfigurationModel>,
+    blob: Attributes<SlackConfigurationModel>
+  ) {
+    super(SlackConfigurationModel, blob);
+  }
+
+  static async makeNew({
+    slackTeamId,
+    connectorId,
+    transaction,
+  }: {
+    slackTeamId: string;
+    connectorId: ModelId;
+    transaction: Transaction;
+  }) {
+    const otherSlackConfigurationWithBotEnabled =
+      await SlackConfigurationModel.findOne({
+        where: {
+          slackTeamId,
+          botEnabled: true,
+        },
+        transaction,
+      });
+
+    await SlackConfigurationModel.create(
+      {
+        slackTeamId,
+        botEnabled: otherSlackConfigurationWithBotEnabled ? false : true,
+        connectorId: connectorId,
+      },
+      { transaction }
+    );
+  }
+
+  static async fetchByConnectorId(connectorId: ModelId) {
+    const blob = await this.model.findOne({
+      where: {
+        connectorId: connectorId,
+      },
+    });
+    if (!blob) {
+      return null;
+    }
+
+    // Use `.get` to extract model attributes, omitting Sequelize instance metadata.
+    return new this(this.model, blob.get());
+  }
+
+  static async fetchByActiveBot(slackTeamId: string) {
+    const blob = await this.model.findOne({
+      where: {
+        slackTeamId,
+        botEnabled: true,
+      },
+    });
+    if (!blob) {
+      return null;
+    }
+
+    // Use `.get` to extract model attributes, omitting Sequelize instance metadata.
+    return new this(this.model, blob.get());
+  }
+
+  async isBotWhitelisted(botName: string | string[]): Promise<boolean> {
```

Comment:
Probably worth adding a funciton to add a slack bot whitelist even if not yet used? (I presume the only way after this PR will be to add it in DB directly?)